server/viame_tasks/tasks.py

```diff
@@ -15,8 +15,52 @@
 from typing import Dict, List
 
 
+def get_available_gpus() -> Dict[str, str]:
+    """
+    Return the list of available GPUs.
+
+    The return of this function is a dictionary that maps each GPU's UUID to its index.
+    """
+    # Get info from nvidia-smi, in the format: index, gpu-uuid
+    # Use this format instead of list due to required equal signs
+    command = "nvidia-smi --query-gpu=index,uuid --format=csv,noheader,nounits"
+
+    process = Popen(command, stderr=PIPE, stdout=PIPE, shell=True)
+    stdout_bytes, _ = process.communicate()
+
+    if process.returncode != 0:
+        return {}
+
+    # Create a list of the form [..., [index, uuid], ...]
+    stdout_lines = [line.strip() for line in stdout_bytes.decode().split("\n") if line]
+    stdout_tuples: List[List] = [line.split(", ") for line in stdout_lines]
+
+    uuid_to_index_dict = {uuid: index for index, uuid in stdout_tuples}
+    return uuid_to_index_dict
```

Comment:
Can we use  https://github.com/anderskm/gputil#main-functions instead?   subprocess and string parsing is OK as a last resort, but that tool will support a lot of what will need in the future without adding more code.

```
import GPUtil
deviceIDs = GPUtil.getAvailable(order = 'first', limit = 1, maxLoad = 0.5, maxMemory = 0.5, includeNan=False, excludeID=[], excludeUUID=[])
```

Reply:
On a related note, yes we should probably also switch to https://github.com/kkroening/ffmpeg-python